### DISCUSSION TITLE
Can I update dependencies my self? If yes, how?

### DISCUSSION THREAD

**Discussion by fischll** [quarto-dev/quarto-cli#8742] [Q&A] [answered]
I would like to use new features of mermaid 10.8 but I can't find a way to upgrade it from mermaid version 10.2 that is curently used in quarto 1.4.
Instead of opening a feature request or an issue I would like to aks the underlying question: How can I update any dependency?

Sorry if this is a trivial question.

**Comment by cscheid** [comment, marked as answer]
It's not something doable from outside quarto right now.

**Reply by fischll** [reply]
Thanks for the quick reply.
Should I create a feature request then?

**Reply by cscheid** [reply]
That's not going to be a priority for us in the near future given our other constraints.

**Reply by mcanouil** [reply]
Side note: if the ability to upgrade components is to be added, all other components would need the same, not just mermaid.

**Comment by neon-ninja** [comment]
You can just stick a `<script>` tag in your qmd file, and that should work just fine. Here's an example: https://github.com/jensbri/QuartoSlides/blob/2a086a490d61382feec64e671dcc2691a1e87d3b/DataWorkflows2024.qmd#L93 & rendered version - https://jensbri.github.io/QuartoSlides/DataWorkflows2024#/placeholder

**Reply by putzwasser** [reply]
Do you have a similar trick for generating PDFs? I'd need to bump mermaidjs to v11.2

**Reply by cderv** [reply]
Mermaid has been updated in 1.6.12 pre-release
- https://github.com/quarto-dev/quarto-cli/pull/10797

Just installed the latest pre-release build if you don't want to wait for next stable release
quarto.org/docs/download/prerelease.html

**Reply by kompre** [reply]
Following your advice I was able to render correctly to html a mermaid block on the latest version (11.12.0). 

Can this workaround be extended to pdf rendering? I tried but I get an error (`ERROR: Internal Error: Couldn't find an svg element in svg string`)

**Reply by cderv** [reply]
> Can this workaround be extended to pdf rendering? I tried but I get an error (ERROR: Internal Error: Couldn't find an svg element in svg string)

Look for this issue in the repo or discussion board: https://github.com/search?q=repo%3Aquarto-dev%2Fquarto-cli+Couldn%27t+find+an+svg+element+in+svg+string&type=issues

This is related to chrome version.
